example_plugins/pair_plugin/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+
+
+set(COMPONENT_NAME pair_plugin)
+
+set(_${COMPONENT_NAME}_sources
+    module.cc
+    )
+
+set(_${COMPONENT_NAME}_cu_sources
+    PotentialPairExampleGPUKernel.cu
+    )
+
+if (ENABLE_HIP)
+set(_cuda_sources ${_${COMPONENT_NAME}_cu_sources})
+endif (ENABLE_HIP)
+
+pybind11_add_module(_${COMPONENT_NAME} SHARED ${_${COMPONENT_NAME}_sources} ${_cuda_sources} NO_EXTRAS)
+# alias into the HOOMD namespace so that plugins and symlinked components both work
+add_library(HOOMD::_${COMPONENT_NAME} ALIAS _${COMPONENT_NAME})
+
+if (APPLE)
+set_target_properties(_${COMPONENT_NAME} PROPERTIES INSTALL_RPATH "@loader_path/..;@loader_path")
+else()
+set_target_properties(_${COMPONENT_NAME} PROPERTIES INSTALL_RPATH "\$ORIGIN/..;\$ORIGIN")
+endif()
+
+# link the library to its dependencies
+target_link_libraries(_${COMPONENT_NAME}
+                      PUBLIC HOOMD::_hoomd
+                      PUBLIC HOOMD::_md
+                      )
+
+fix_cudart_rpath(_${COMPONENT_NAME})
+
+# install the library
+install(TARGETS _${COMPONENT_NAME}
+        LIBRARY DESTINATION ${PYTHON_SITE_INSTALL_DIR}/${COMPONENT_NAME}
+        )
+
+################ Python only modules
+# copy python modules to the build directory to make it a working python package
+set(files
+    __init__.py
+    pair.py
+    )
+
+install(FILES ${files}
+        DESTINATION ${PYTHON_SITE_INSTALL_DIR}/${COMPONENT_NAME}
+       )
+
+copy_files_to_build("${files}" "${COMPONENT_NAME}" "*.py")
+
+if (BUILD_TESTING)
+    # add_subdirectory(test-py)
```

Comment:
Sure thing! Do you think just python tests are sufficient? No need to write C++ tests for these simple cases?

Reply:
Yes, Python tests are sufficient.

Reply:
Hi Joshua, I added the pytests. But like you said, the namespace is currently inconsistent in the build directory. For external components this behaviour is okay, but at the moment this is breaking in-build testing when building hoomd itself since example_plugins is being symlinked into the hoomd dir. Do you have any suggestions?

Reply:
Let's ignore the plugin tests in the build directory to get around this.

Add a `pytest.ini` file that gets copied into the directory `build/hoomd` with the contents:
```
[pytest]
norecursedirs = *_plugin
```
To copy it to the build directory, add `pytest.ini` to the `set(files` list in `hoomd/CMakeLists.txt`.